site-modules/p9_instance_setup/plans/upload_auth_files.pp

```diff
@@ -0,0 +1,48 @@
+#  Uploads authentication files known to be needed for running performance testing workflows
+#
+#  options:
+#   instance_username - name of user to setup
+#
+#  examples:
+#    run_plan(p9_instance_setup::upload_auth_files,'nodes' => $nodes, instance_username => $username)
+#
+#    bolt plan run p9_instance_setup::upload_auth_files --nodes 10.234.0.19 --user centos 
+#      --private-key ~/.ssh/id_rsa-acceptance --no-host-key-check --tty --run-as root instance_username=betty
+#
+plan p9_instance_setup::upload_auth_files (
+     TargetSpec $nodes,
+     String $instance_username = "",
+   ) {
+  
+  $username = run_plan(p9_instance_setup::determine_username,'nodes' => "localhost",
+    instance_username => $instance_username)
+  notice("Username will be $username")
+
+  $result_pwd = run_command('echo $HOME',"localhost",'_catch_errors' => true)
+  $home_dir = $result_pwd.first.value[stdout].strip
+
+  # abs, vmpooler, aws account access 
+  $fog_file_src = "$home_dir/.fog"
+  $fog_file_dest = "/home/$username/.fog"
+  if p9_instance_setup::readable($fog_file_src) {
+    upload_file($fog_file_src,$fog_file_dest,$nodes,'_run_as' => "$username")
+  }
+
+  # for aws s3 access
+  $aws_cred_file_src = "$home_dir/.aws/credentials"
+  $aws_cred_file_dest = "/home/$username/.aws/credentials"
+  if p9_instance_setup::readable($aws_cred_file_src) {
+    run_command("mkdir -p /home/$username/.aws",$nodes,'_run_as' => "$username")
+    upload_file($aws_cred_file_src,$aws_cred_file_dest,$nodes,'_run_as' => "$username")
+  }
+
+  #private a public keys needed to login to vmpooler, aws, and other p9 instances
```

Comment:
Personally, I would rather supply a list of specific keys to copy rather than ex-filtrating all of my keys to a remote node. Ticketed as https://tickets.puppetlabs.com/browse/SLV-442 for future modification.

Reply:
ok, we can discuss that one in grooming.